api/migrations/20230302152346_results-of-prev-activities-mmis.js

```diff
@@ -0,0 +1,87 @@
+import loggerFactory from '../logger/index.js';
+import { setup, teardown } from '../db/mongodb.js';
+import { MMIS } from '../models/index.js';
+
+const logger = loggerFactory(
+  'mongoose-migrate/migrate-results-of-previous-activities-mmis'
+);
+
+/**
+ * Update Results of Previous Activities page in MMIS APDs to have DDI and M&O
+ */
+export const up = async () => {
+  //Grab all APDs
+  await setup();
+  const apds = await MMIS.find().lean();
+  logger.info(`Updating ${apds.length} APDs`);
+
+  const updatedApds = apds.map(apd => {
+    const years = apd.years;
+    let prevYears = [];
+
+    [0, 1, 2].map(past => prevYears.unshift(years[0] - past));
+
+    if (apd.previousActivities.actualExpenditures[prevYears[0]].ddi) {
+      return apd;
+    }
+
+    return {
+      ...apd,
+      id: apd._id,
+      previousActivities: {
+        ...apd.previousActivities,
+        actualExpenditures: prevYears.reduce(
+          (acc, year) => ({
+            ...acc,
+            [year]: {
+              ddi: {
+                50: {
+                  federalActual: 0,
```

Comment:
Is the migration script supposed to populate the value of the previous activity in here or is it just to create the structure and have it all zeroed out?

Reply:
I believe create the structure and zero it out. Since the previous values would have the wrong structure so their actual values would be irrelevant